conans/test/unittests/tools/cmake/test_cmake_install.py

```diff
@@ -0,0 +1,76 @@
+import platform
+import pytest
+
+from conan.tools.cmake import CMake
+from conan.tools.cmake.presets import write_cmake_presets
+from conans.client.conf import get_default_settings_yml
+from conans.model.conf import Conf
+from conans.model.settings import Settings
+from conans.test.utils.mocks import ConanFileMock
+from conans.test.utils.test_files import temp_folder
+
+
+@pytest.mark.parametrize("generator", [
```

Comment:
I appreciate the thorough testing, but it is really necessary to test with all CMake generators?
The logic for ``--component`` doesn't seem tied to the generator at all, so maybe this can be simplified a bit.

Reply:
Probably not needed, I originally put it in because I was unsure if the implementation would need to change based on the generator but later determined it would not. I will simplify when I can.